### PR TITLE
Add platform-specific props to wrapped Modal.

### PR DESCRIPTION
**Pull request by TheTekton**
Platform-specific Modal props are needed to support device orientation on iOS. Also adding the onShow func prop, in case a use-case arises.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,13 +60,31 @@ class Overlay extends React.Component {
...
-          animationType={animationType}
-          transparent
-          visible={this.state.visible}
-          onRequestClose={this._hideModal}>
+        animationType={animationType}
+        transparent={transparent}
+        visible={this.state.visible}
+        onRequestClose={this._hideModal}
+        onShow={onShow}
+        hardwareAccelerated={hardwareAccelerated}
```
Comment (rgabs) [inline index.js]: I think a better way would be to forward all the extra props to the modal component.

Something like 

```
const {animationType, closeOnTouchOutside, children, containerStyle, childrenWrapperStyle, ...extraProps} = this.props;
```

Then pass the extraProps to the Modal component like this:

```
<Modal {...extraProps}>
```

**Comment by TheTekton** [comment]
I'll push an update for props by this weekend.

**Comment by TheTekton** [comment]
@rgabs I think that's set now. A bit tired, but hopefully that covers it.

**Comment by rgabs** [comment]
LGTM. Thanks for the contribution @TheTekton 
Cheers 🍻

**Pull request merged**